test/cypress/plugins/stubs.js

```diff
@@ -140,5 +141,76 @@ module.exports = {
     const path = `/v1/frontend/charges/${opts.chargeId}/capture`
 
     return simpleStubBuilder('POST', 204, path, undefined)
+  },
+
+  connectorWorldPay3dsFlexDdcJwt: (opts = {}) => {
+    const path = `/v1/frontend/charges/${opts.chargeId}/worldpay/3ds-flex/ddc`
+
+    const body = worldPay3dsFlexDdcJwtFixtures.validDdcJwt().getPlain()
+
+    return simpleStubBuilder('GET', 200, path, body)
+  },
+
+  worldpay3dsflexddcIframePost: (opts = {}) => {
+    const body = `<!DOCTYPE html>
+    <html>
+    <head>
+        <title>Cardinal DDC Sim</title>
+    </head>
+    <body>
+    <script>
+        sendNotification(true, "${opts.sessionId}");
+
+        function sendNotification(status, sessionId){
+            try{
+                var message = {
+                    MessageType: 'profile.completed',
+                    SessionId: sessionId,
+                    Status: status
+                };
+                window.parent.postMessage(JSON.stringify(message), '*');
+            } catch(error){
+                console.error('Failed to notify parent', error)
+            }
+        }
+    </script>
+    </body>
+    </html>`
+
+    return [{
+      predicates: [{
+        equals: {
+          method: 'POST',
+          path: '/shopper/3ds/ddc.html'
+        }
+      }],
+      responses: [{
+        is: {
+          statusCode: 200,
+          headers: {
+            'Content-Type': 'text/html;charset=ISO-8859-1'
```

Comment:
I assume this comes from what the Worldpay DDC simulator actually uses, in which case someone needs to remind them it’s 2019!

Reply:
ha yes

Reply:
lol